pkg/state/state_test.go

```diff
@@ -3455,3 +3455,19 @@ func TestAppendChartDownloadTLSFlags(t *testing.T) {
 		})
 	}
 }
+
+func TestHideChartURL(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected string
+	}{
+		{"http://username:password@example.com/", "http@example.com/"},
+		{"http://example.com@", "http@"},
```

Comment:
more cases:  
1. https://username:password@example.com/
2. https://username:@password@example.com/
3. https://username::password@example.com/
4. https://username:httpd@example.com/
5. https://username:httpsd@example.com/

Reply:
Done! Also made a improvement to the function I created, thanks @yxxhero!